File: 0003/3.cs
public class Solution {
    public int LongestSubstringWithoutRepeati(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(LongestSubstringWithoutRepeati(root.left), LongestSubstringWithoutRepeati(root.right));
    }
}

File: 0003/3.go
func longestSubstringWithoutRepeati(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := longestSubstringWithoutRepeati(root.Left)
	r := longestSubstringWithoutRepeati(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}